[WIP] This PR is designed to update Posrednik to the latest version, with various tweaks and fixes

YorkAARGH opened the pull request:
Updated to recent Klasa
Updated modlog config to use `channels` folder
Added antiinvite boolean to klasaReady event
Added `channels` folder

YorkAARGH committed:
Updating phase 1

Update bot to latest version of klasa, and general update and improvement of code.

YorkAARGH committed:
Tyler Fix

Part 1 of many?

Tylertron1998 committed:
should work

YorkAARGH committed:
Merge pull request #1 from Tylertron1998/patch-1

should work

YorkAARGH committed:
Fixed modlogs

Persistent modlogs in guild schema

YorkAARGH committed:
Update

Updated moderation commands to new role hierarchy methods.
Changed softban, warn and kick to member instead of user.
Updated reason, case and history command to new provider methods.

bdistin commented on commands/Moderation/case.js:
```diff
@@ -16,18 +16,22 @@ module.exports = class extends Command {
 
     init() {
-        this.provider = this.client.providers.get('json');
+        this.provider = this.client.providers.default;
```
while I understand that this was like this to begin with. It's best to avoid this since that locks a provider into this command, and prevents you from reloading it in the context of this command. Better to just directly use this.client.providers.default. Also, I don't see this.provider being used in this window, if it's unused, you can remove the init method completely.

bdistin commented on commands/Moderation/history.js:
```diff
@@ -36,7 +35,7 @@ module.exports = class extends Command {
 
     init() {
-        this.provider = this.client.providers.get('json');
+        this.provider = this.client.providers.default;
```
same

bdistin commented on commands/Moderation/softban.js:
```diff
@@ -11,37 +11,35 @@ module.exports = class extends Command {
     }
 
-    async run(msg, [user, ...reason]) {
+    async run(msg, [member, days = 1, ...reason]) {
```
correct me if I am wrong, but I think this was a user to still softban someone even if they left the guild, allowing the messages to be cleared.

YorkAARGH replied:
I did not consider that, I will revert that change.

bdistin commented on posrednik.js:
```diff
@@ -5,14 +5,14 @@ const config = require('./config.json');
-    .addLevel(4, false, (client, msg) => msg.guild && msg.member.permissions.has('BAN_MEMBERS'))
-    .addLevel(6, false, (client, msg) => msg.guild && msg.member.permissions.has('ADMINISTRATOR'))
+    .add(0, () => true)
+    .add(2, (client, msg) => msg.guild && msg.member.permissions.has('MANAGE_GUILD'))
```
add { fetch: true } option to autofetch the member if uncached on levels 2-6

bdistin commented on util/modlog.js:
```diff
@@ -95,7 +89,7 @@ module.exports = class ModLog {
 
     get provider() {
-        return this.client.providers.get('json');
+        return this.client.providers.default;
```
same with first comments.

YorkAARGH committed:
Fixed a lot of mistakes

A lof of settings weren't added to any init method or a custom `klasaReady` event.
Also removed rogue newlines from command constructors.

All commands have been updated to utilize the new language methods, all commands confirmed working.

YorkAARGH committed:
Requested changes

Fixed requested changes

YorkAARGH committed:
Missed one...

YorkAARGH committed:
Minior tweaks

Made the action type's first letter upper case (looks better)
Made the warn command require the reason instead of having it as optional.

YorkAARGH committed:
Go to hell codacy...

MrJacz commented on posrednik.js:
```diff
@@ -21,11 +21,12 @@ class Posrednik extends Client {
             // Grabs the newly set permission levels from up top
             permissionLevels
         });
+        this.methods = { Embed: require('discord.js').MessageEmbed };
```
1st. `methods` was removed from client for a reason, the prop isnt related to the instance of the client it doesnt make sense to have it assigned to the instance.
2nd. discord.js should be required outside the constructor

YorkAARGH replied:
Fixed

MrJacz commented on posrednik.js:
```diff
@@ -21,11 +21,12 @@ class Posrednik extends Client {
         super({
             clientOptions: {},
```
clientOptions doesnt exist

YorkAARGH replied:
Fixed

MrJacz commented on posrednik.js:
```diff
@@ -5,14 +5,14 @@ const config = require('./config.json');
     // These 2 levels define the bot owners permission levels
-    .addLevel(9, true, (client, msg) => msg.author === client.owner)
-    .addLevel(10, false, (client, msg) => msg.author === client.owner);
+    .add(9, (client, msg) => msg.author === client.owner)
```
you're missing a `{ break: true }`

YorkAARGH replied:
Fixed

MrJacz commented on events/klasaReady.js:
```diff
@@ -0,0 +1,27 @@
+
+    async init() {
+        // Folders
+        if (!this.client.gateways.guilds.schema.has('channels')) await this.client.gateways.guilds.schema.add('channels', {});
```
All these should be in the related files, such as announcement for roles.announcementRole

YorkAARGH replied:
I won't be doing this change as it would lead to slight code duplication (checking if the schema has channels) for all related files.

I feel doing this check in the klasaReady event is the best course of action.

MrJacz commented on util/modlog.js:
```diff
@@ -58,46 +58,32 @@ module.exports = class ModLog {
 
-    // Here we get the case number and create a modlog provider entry
-
     async getCase() {
```
useless

YorkAARGH replied:
I accidentally removed that, I've put it back in.

MrJacz replied:
useles async...

MrJacz commented on util/modlog.js:
```diff
@@ -58,46 +58,32 @@ module.exports = class ModLog {
-        await this.provider.replace('modlogs', this.guild.id, modlogs);
-        return modlogs.length - 1;
+        this.case = this.guild.configs.modlogs.length;
+        await this.guild.configs.update('modlogs', this.pack);
```
return its already a promise

YorkAARGH replied:
so just straight up `return` instead of `await` ?

YorkAARGH committed:
Some requested changes

YorkAARGH committed:
Updated to latest klasa

Updated permission levels.
Fixed permission levels for commands.
